lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -224,6 +228,13 @@ def from_dir_path(  # type: ignore[misc]
         if dir_path.endswith(to_bytes(os.path.sep)):
             dir_path = dir_path.rstrip(to_bytes(os.path.sep))
         if not _is_collection_dir(dir_path):
+            dir_pathlib = pathlib.Path(to_text(dir_path))
+
+            # special handling for bundled collections without manifests, e.g., ansible._protomatter
+            if dir_pathlib.is_relative_to(_ANSIBLE_PACKAGE_PATH):
+                req_name = f'{dir_pathlib.parent.name}.{dir_pathlib.name}'
+                return cls(req_name, ansible.release.__version__, dir_path, 'dir', None)
+
```

Comment:
Can we move the check to a function and reduce nesting for the sake of readability and preserving the existing style?
```suggestion
        dir_pathlib = pathlib.Path(to_text(dir_path))
        if _is_stdlib_collection_dir(dir_pathlib):
            # special handling for bundled collections without manifests, e.g., ansible._protomatter
            req_name = f'{dir_pathlib.parent.name}.{dir_pathlib.name}'
            return cls(req_name, ansible.release.__version__, dir_path, 'dir', None)
        if not _is_collection_dir(dir_path):
```

Reply:
Although that would reduce nesting, doing so means we would be unnecessarily checking each path to see if it is a builtin collection, rather than only the ones which are known not to be a collection dir. Since we're only performing this check once, there doesn't appear to be any benefit to factoring out the conditional into a function.